Tests/ControllersTests/PartialContentControllerTest.swift

```diff
@@ -0,0 +1,60 @@
+import XCTest
+@testable import Requests
+@testable import Controllers
+
+class PartialContentControllerTest: XCTestCase {
+  func testItCanRespondWithA206WithPartialContent() {
+    let rawRequest = "GET /partial_content.txt HTTP/1.1\r\n Host: localhost:5000\r\n Connection: Keep-Alive\r\n User-Agent: Apache-HttpClient/4.3.5 (java 1.5)\r\n Accept-Encoding: gzip,deflate\r\nRange:bytes=4-"
+    let request = Request(for: rawRequest)
+
+    let contents = ["partial_content.txt": Array("This is a file that contains text to read part of in order to fulfill a 206.".utf8)]
+
+    PartialContentController.setData(contents: contents)
+    let response = PartialContentController.process(request)
+
+    XCTAssertEqual(response.statusCode, "206 Partial Content")
+  }
+
+  func testItCanRespondWithPartialContentsGivenRangeStart() {
+    let rawRequest = "GET /partial_content.txt HTTP/1.1\r\n Host: localhost:5000\r\n Connection: Keep-Alive\r\nRange:bytes=4-\r\n User-Agent: Apache-HttpClient/4.3.5 (java 1.5)\r\n Accept-Encoding: gzip,deflate"
+    let request = Request(for: rawRequest)
+
+    let contents = ["partial_content.txt": Array("This is a file that contains text to read part of in order to fulfill a 206.".utf8)]
+
+    PartialContentController.setData(contents: contents)
+    let response = PartialContentController.process(request)
+
+    let expected = Array("\n\n is a file that contains text to read part of in order to fulfill a 206.".utf8)
+
+    XCTAssertEqual(response.body!, expected)
+  }
+
+  func testItCanRespondWithPartialContentsGivenRangeEnd() {
+    let rawRequest = "GET /partial_content.txt HTTP/1.1\r\n Host: localhost:5000\r\n Connection: Keep-Alive\r\nRange:bytes=-6\r\n User-Agent: Apache-HttpClient/4.3.5 (java 1.5)\r\n Accept-Encoding: gzip,deflate"
+    let request = Request(for: rawRequest)
+
+    let contents = ["partial_content.txt": Array("This is a file that contains text to read part of in order to fulfill a 206.".utf8)]
+
+    PartialContentController.setData(contents: contents)
+    let response = PartialContentController.process(request)
+
+    let expected = Array("\n\na 206.".utf8)
+
+    XCTAssertEqual(response.body!, expected)
+  }
+
+  func testItCanRespondWithPartialContentsGivenRangeStartAndEnd() {
+    let rawRequest = "GET /partial_content.txt HTTP/1.1\r\n Host: localhost:5000\r\n Connection: Keep-Alive\r\nRange:bytes=0-4\r\n User-Agent: Apache-HttpClient/4.3.5 (java 1.5)\r\n Accept-Encoding: gzip,deflate"
+    let request = Request(for: rawRequest)
+
+    let contents = ["partial_content.txt": Array("This is a file that contains text to read part of in order to fulfill a 206.".utf8)]
+
+    PartialContentController.setData(contents: contents)
+    let response = PartialContentController.process(request)
```

Comment:
Similar to the problem above, I'm grouping the controllers that are dependent on file data in a `FileDataController` protocol in order to inject mocks for testing - but now I have to use this setup method before I can use the function I want.

Is this a necessary evil? I know in the Protocol Oriented Programming video that the presenter mentioned that protocols provide a better way to mock your tests...but I'm not exactly sure I understand how.